src/timelock.cairo

```diff
@@ -160,26 +273,27 @@ mod Timelock {
             results
         }
 
-        fn get_execution_window(self: @ContractState, id: felt252) -> (u64, u64) {
-            let start_time = self.execution_started.read(id);
+        fn get_execution_window(self: @ContractState, id: felt252) -> ExecutionWindow {
+            let start_time = self.execution_state.read(id).started;
 
             // this is how we prevent the 0 timestamp from being considered valid
             assert(start_time != 0, 'DOES_NOT_EXIST');
```

Comment:
hmmm maybe we could just check that canceled is nonzero as well and remove the other logic changes. sorry, i realize this is out of scope for the initial pr.